frontend/src/component/MenuList/NavGroup/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { ReactNode } from 'react';
+import { Divider, List, Typography } from '@mui/material';
+
+// material-ui
+import { GenericCardProps } from '../../../types';
+// eslint-disable-next-line import/no-cycle
+import NavCollapse from '../NavCollapse';
+// project imports
+import NavItem from '../NavItem';
+
+export interface NavGroupProps {
+  item: {
+    id?: string;
+    type?: string;
+    icon?: GenericCardProps['iconPrimary'];
+    children?: NavGroupProps['item'][];
+    title?: ReactNode | string;
+    caption?: ReactNode | string;
+    color?: 'primary' | 'secondary' | 'default' | undefined;
```

Comment:
**suggestion:** Create enum for color, I'll mention this bit later in the end again.